py/server/deephaven/table.py

```diff
@@ -896,6 +915,7 @@ def group_by(self, by: Union[str, Sequence[str]] = None) -> Table:
         except Exception as e:
             raise DHError(e, "table group operation failed.") from e
 
+    @auto_locking_op
     def ungroup(self, cols: Union[str, Sequence[str]] = None) -> Table:
```

Comment:
`format*` locks?

Reply:
layout hints?

Reply:
`PartitionedTable.merge` lock?

Reply:
`PartitionedTable.filter`, `PartitionedTable.sort`, `PartitionedTable.transform`, maybe other PartitionedTable methods need lock?

Reply:
No aggs need locks.
No format or layout methods need locks.
Partition by does not need a lock.
PartitionedTable filter and sort do not need locks.

PartitionedTable merge (because it has multiple tables), transform (because it uses update), and partitioned transform (because it uses update and joins) do need locks.